test/test_image.py

```diff
@@ -100,14 +100,15 @@ def test_decode_jpeg(img_path, pil_mode, mode):
     assert abs_mean_diff < 2
 
 
+@pytest.mark.parametrize("codec", [("png", "PNG"), ("jpg", "JPEG")])
 @pytest.mark.parametrize("orientation", [1, 2, 3, 4, 5, 6, 7, 8, 0])
-def test_decode_jpeg_with_exif_orientation(tmpdir, orientation):
-    fp = os.path.join(tmpdir, f"exif_oriented_{orientation}.jpg")
+def test_decode_with_exif_orientation(tmpdir, codec, orientation):
```

Comment:
Should we also add a png test for `test_invalid_exif()` ?

Reply:
no, png does not accept exif data as "111111...." and fails on saving the file.